tests/shtest

```diff
@@ -275,11 +275,19 @@ if [ -n "$($VALGRIND $Q $JQ -n 'halt_error(1)' 2>&1)" ]; then
     echo "jq halt_error(1) had unexpected output" 1>&2
     exit 1
 fi
-if [ -n "$($VALGRIND $Q $JQ -n '"xyz\n"|halt_error(1)' 2>/dev/null)" ]; then
+if [ -n "$($VALGRIND $Q $JQ -n '"xyz\n" | halt_error(1)' 2>/dev/null)" ]; then
     echo "jq halt_error(1) had unexpected output on stdout" 1>&2
     exit 1
 fi
-if [ "$($VALGRIND $Q $JQ -n '"xyz\n"|halt_error(1)' 2>&1)" != "jq: error: xyz" ]; then
+if [ "$($VALGRIND $Q $JQ -n '"xy" | halt_error(1)' 2>&1 || echo "z")" != "xyz" ]; then
+    echo "jq halt_error(1) had unexpected output" 1>&2
+    exit 1
+fi
+if [ "$($VALGRIND $Q $JQ -n '"x\u0000y\u0000z" | halt_error(1)' 2>&1)" != "$(printf 'x\x00y\x00z')" ]; then
+    echo "jq halt_error(1) had unexpected output" 1>&2
+    exit 1
```

Comment:
Seems linux build is not happy about this:
```
+ /home/runner/work/jq/jq/tests/../jq -n halt_error(1)
+ [ 1 -ne 1 ]
+ /home/runner/work/jq/jq/tests/../jq -n halt_error(11)
+ [ 11 -ne 11 ]
+ /home/runner/work/jq/jq/tests/../jq -n halt_error(1)
+ [ -n  ]
+ /home/runner/work/jq/jq/tests/../jq -n "xyz\n" | halt_error(1)
+ [ -n  ]
+ /home/runner/work/jq/jq/tests/../jq -n "xy" | halt_error(1)
+ echo z
+ [ xyz != xyz ]
+ /home/runner/work/jq/jq/tests/../jq -n "x\u0000y\u0000z" | halt_error(1)
+ printf x\x00y\x00z
+ [ xyz != x\x00y\x00z ]
+ echo jq halt_error(1) had unexpected output
jq halt_error(1) had unexpected output
+ exit 1
```
Hmm zero bytes gone from LHS value? shell doing something?

Reply:
Yeah, NUL bytes are trimmed in command substitution.

Reply:
Aha, any idea why macOS build was fine?

Reply:
bash is old. It passes if Bash is 3.

Reply:
👍 that explains things, great work!

Reply:
If necessary, `\0` can be used instead of `\x00`. That would work in any version bash or POSIX-compliant shell.